### ISSUE TITLE
T-selector

### ISSUE THREAD

**Issue by qyc-98** [Hon-Wong/Elysium#12]
Hi, thanks for your wonderful work!
I'd like to ask how to implement the Gumbel softmax in Tselector for T-selector's training.( in this code)
![image](https://github.com/user-attachments/assets/e667304c-6cdd-42bd-b2e8-779da0fb42ec)

My implementations are:

![image](https://github.com/user-attachments/assets/f11c385d-52d3-4c93-ae80-dddec19d8dc3)

**Comment by Hon-Wong** [comment]
Most are correct, but "hard" should be set as False, , as indicated [here](https://pytorch.org/docs/stable/generated/torch.nn.functional.gumbel_softmax.html#:~:text=hard%20(bool)%20%E2%80%93%20if%20True%2C%20the%20returned%20samples%20will%20be%20discretized%20as%20one%2Dhot%20vectors%2C%20but%20will%20be%20differentiated%20as%20if%20it%20is%20the%20soft%20sample%20in%20autograd)

We simply replace the softmax operation with:
`video_token_scores = F.gumbel_softmax(video_token_logits, tau=1, hard=False)`
